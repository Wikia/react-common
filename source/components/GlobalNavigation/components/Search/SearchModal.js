import React from 'react';
import PropTypes from 'prop-types';

import Button from '../../../Button';
import Icon from '../../../Icon';

const SearchModal = ({ openModal, children, isOpen }) => (
    <React.Fragment>
        <Button
            className="wds-global-navigation__modal-control wds-global-navigation__modal-control-search"
            onClick={() => openModal('search')}
            text
        >
            <Icon name="magnifying-glass" />
        </Button>
        {
            isOpen && (
                <div className="wds-global-navigation__modal wds-search-modal">
                    <div className="wds-search-modal__form-and-links-wrapper">
                        {children}
                    </div>
                </div>
            )
        }
    </React.Fragment>
);

SearchModal.propTypes = {
    children: PropTypes.node.isRequired,
    isOpen: PropTypes.bool.isRequired,
    openModal: PropTypes.func.isRequired,
};

export default SearchModal;
