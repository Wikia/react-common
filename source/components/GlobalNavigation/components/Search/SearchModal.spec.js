import React from 'react';
import { shallow } from 'enzyme';
import merge from 'lodash/merge';

import SearchModal from './SearchModal';

const defaultProps = {
    openModal: () => null,
    children: <div>YES, I AM A CHILD</div>,
    isOpen: false,
};

function renderComponent(props) {
    const computedProps = merge({}, defaultProps, props);

    return shallow(<SearchModal {...computedProps} />);
}

test('SearchModal renders correctly when isOpen is false', () => {
    expect(renderComponent()).toMatchSnapshot();
});

test('SearchModal renders correctly when isOpen is true', () => {
    expect(renderComponent({ isOpen: true })).toMatchSnapshot();
});

test('SearchModal calls openModal onClick', () => {
    const openModalSpy = jest.fn();
    const wrapper = renderComponent({ openModal: openModalSpy });

    wrapper.find('Button').props().onClick();

    expect(openModalSpy).toBeCalledWith('search');
});
