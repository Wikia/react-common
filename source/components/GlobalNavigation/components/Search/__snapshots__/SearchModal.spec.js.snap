// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchModal renders correctly when isOpen is false 1`] = `
<Fragment>
  <Button
    className="wds-global-navigation__modal-control wds-global-navigation__modal-control-search"
    disabled={false}
    fullwidth={false}
    href={null}
    light={false}
    onClick={[Function]}
    secondary={false}
    square={false}
    text={true}
  >
    <Icon
      className=""
      name="magnifying-glass"
      small={false}
      tiny={false}
    />
  </Button>
</Fragment>
`;

exports[`SearchModal renders correctly when isOpen is true 1`] = `
<Fragment>
  <Button
    className="wds-global-navigation__modal-control wds-global-navigation__modal-control-search"
    disabled={false}
    fullwidth={false}
    href={null}
    light={false}
    onClick={[Function]}
    secondary={false}
    square={false}
    text={true}
  >
    <Icon
      className=""
      name="magnifying-glass"
      small={false}
      tiny={false}
    />
  </Button>
  <div
    className="wds-global-navigation__modal wds-search-modal"
  >
    <div
      className="wds-search-modal__form-and-links-wrapper"
    >
      <div>
        YES, I AM A CHILD
      </div>
    </div>
  </div>
</Fragment>
`;
